packer/scripts/20-caddy.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+groupadd --system caddy
+useradd --system \
+    --gid caddy \
+    --create-home \
+    --home-dir /var/lib/caddy \
+    --shell /usr/sbin/nologin \
+    --comment "Caddy web server" \
+    caddy
+
+chown -R caddy:caddy /var/lib/caddy
+
+wget -O /usr/bin/caddy https://github.com/caddyserver/caddy/releases/download/v2.0.0-beta8/caddy2_beta8_linux_amd64
```

Comment:
Not a huge deal, but it'd be good to extract the version number to a variable, which could be fed in through the provisioner (using the [`environment_vars`](https://www.packer.io/docs/provisioners/shell.html#environment_vars) key).

Also to limit terminal output it's probably useful to do `wget -qO` instead:

```suggestion
wget -qO /usr/bin/caddy https://github.com/caddyserver/caddy/releases/download/v2.0.0-beta8/caddy2_beta8_linux_amd64
```

Reply:
Done